### PR TITLE
Remove mysql/postgres from extras before determining the installation_command_flags

### PR DESCRIPTION
**Pull request by macchiang**
- Fix the build error with the build-args `INSTALL_MYSQL_CLIENT=false` or `INSTALL_POSTGRES_CLIENT=false`  

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice! Thanks!

**Comment by potiuk** [comment]
SOme static check failure. The other issue with tests is being fixed (broken main)

File: Dockerfile [hunk 1]
```diff
@@ -1456,7 +1456,7 @@ ARG AIRFLOW_USER_HOME_DIR
 ARG AIRFLOW_UID
 
 RUN adduser --gecos "First Last,RoomNumber,WorkPhone,HomePhone" --disabled-password \
-       --quiet "airflow" --uid "${AIRFLOW_UID}" --gid "0" --home "${AIRFLOW_USER_HOME_DIR}" && \
+    --quiet "airflow" --uid "${AIRFLOW_UID}" --gid "0" --home "${AIRFLOW_USER_HOME_DIR}" && \
```
Comment (Taragolis) [inline Dockerfile]: Are this multiple indent changes required to this PR?

**Review by dirrao — COMMENTED** [review]
Congratulations on your first PR. 
Are these indentation  changes intentional?  Can you look into the failing checks?

File: Dockerfile (hunk 1 above)
Comment (macchiang) [reply Dockerfile]: No needed. Thanks. 😅

**Comment by macchiang** [comment]
@potiuk @dirrao All fixed.

**Comment by potiuk** [comment]
Still some static-checks failing (I recommend installing and running `pre-commit` locally, it will fix things for you.

**Comment by macchiang** [comment]
@potiuk All good now. 🚀

**Review by Taragolis — APPROVED** [review]

**Pull request merged**